packages/flutter/lib/src/painting/borders.dart

```diff
@@ -294,6 +313,41 @@ class BorderSide {
     );
   }
 
+  /// Get the amount of the stroke width that lies inside of the [BorderSide].
+  ///
+  /// This will return 0 for a [strokeAlign] of 1, and the width of the stroke for
+  /// a [strokeAlign] of -1.
+  ///
+  /// Example of results:
+  /// StrokeAlign.inside (-1) => width
+  /// StrokeAlign.center  (0) => width / 2
+  /// StrokeAlign.outside (1) => 0
+  /// ```
```

Comment:
It's better than the code, but I think just using the words might work better:

```suggestion
  /// Get the amount of the stroke width that lies inside of the [BorderSide].
  ///
  /// For example, this will return 0 for a [strokeAlign] of 1, half the [width] for
  /// a [strokeAlign] of 0, and the [width] for a [strokeAlign] of -1.
```

Reply:
Last try. What do you think?
<img width="684" alt="image" src="https://user-images.githubusercontent.com/351125/182261754-915ff62a-9c4d-401b-944e-79a1344a59cb.png">

Reply:
One line per strokeAlign.

Reply:
Ok, give up. I'm doing what you recommend, but -1 -> 1 (the old was 1 -> -1). I don't understand anything, it is too hard to read. But I don't know how to improve.

Reply:
Those look good now, and -1, 0, 1 seems logical.